File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i=0, j=0, k=0;

        while(i<word1.length() && j<word2.length()){
            if(k%2==0){
                sb.append(word1.charAt(i));
                i++;
            }
            else{
                sb.append(word2.charAt(j));
                j++;
            }

            k++;
        }

        while(i<word1.length()){
            sb.append(word1.charAt(i));
            i++;
        }

        while(j<word2.length()){
            sb.append(word2.charAt(j));
            j++;
        }

        return sb.toString();
    }
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        word = ""
        i = 0
        j = 0

        while i<len(word1) and j<len(word2):
            word = word + word1[i]
            word = word + word2[j]
            i+=1
            j+=1
        
        while i<len(word1):
            word = word + word1[i]
            i+=1

        while j<len(word2):
            word = word + word2[j]
            j+=1

        return word